jacodb-analysis/src/test/kotlin/org/jacodb/analysis/impl/JodaDateTimeAnalysisTest.kt

```diff
@@ -53,37 +64,66 @@ class JodaDateTimeAnalysisTest : BaseTest() {
 
     private val graph: JcApplicationGraph by lazy {
         runBlocking {
-            cp.newApplicationGraphForAnalysis()
+            JcApplicationGraphImpl(cp, cp.usagesExt())
         }
     }
 
     @Test
-    fun `test taint analysis`() {
+    fun `test taint analysis`() = runBlocking {
         val clazz = cp.findClass<DateTime>()
         val methods = clazz.declaredMethods
-        val unitResolver = SingletonUnitResolver
-        val manager = TaintManager(graph, unitResolver)
-        val sinks = manager.analyze(methods, timeout = 60.seconds)
+
+        val ifdsContext = taintIfdsContext(cp, graph, defaultBannedPackagePrefixes)
+        val system = systemOf("ifds") { ProjectManager(ifdsContext) }
+
+        for (method in methods) {
+            system.startTaintAnalysis(method)
+        }
+        launch {
+            delay(20.seconds)
+            system.stop()
+        }
+        system.awaitCompletion()
+        system.resume()
+        val sinks = system.collectTaintResults()
+
         logger.info { "Vulnerabilities found: ${sinks.size}" }
     }
 
     @Test
-    fun `test NPE analysis`() {
+    fun `test NPE analysis`() = runBlocking {
         val clazz = cp.findClass<DateTime>()
         val methods = clazz.declaredMethods
-        val unitResolver = SingletonUnitResolver
-        val manager = NpeManager(graph, unitResolver)
-        val sinks = manager.analyze(methods, timeout = 60.seconds)
+        val ifdsContext = npeIfdsContext(cp, graph, defaultBannedPackagePrefixes)
+        val system = systemOf("ifds") { ProjectManager(ifdsContext) }
+
+        for (method in methods) {
+            system.startNpeAnalysis(method)
+        }
+        launch {
+            delay(20.seconds)
+            system.stop()
+        }
+        system.awaitCompletion()
+        system.resume()
```

Comment:
Why do you need to call `resume` here? Other tests (juliet-sql, for example) do not contain such calls. Is it because of `system.stop()` above? In this case, why not implement the timeout via `launch { 
delay(); stop(); resume() }`, i.e. call `resume` after `stop` only when necessary?